### PR TITLE
Specify `python_requires` in setup.py. Fixes #2942

### LINKED ISSUE #2942: Installing on old pythons.
### Setup

I am reporting a problem with Biopython source, and Python 3.5. This stems from the pypi packages not declaring for which versions of python they are compatible (not specifying `python_requires="....",` in the call to `setup()` in `setup.py`.

### Expected behaviour

Trying to install biopython with pip should install a version of biopython that is compatible with python 3.5.

### Actual behaviour

Without a version specified pip will download the latest and then when `setup.py` is invoked the python version is caught and the install exits.

### Steps to reproduce

```pip install biopython

Collecting biopython
  Downloading biopython-1.77.tar.gz (16.8 MB)
     |████████████████████████████████| 16.8 MB 9.5 MB/s
    ERROR: Command errored out with exit status 1:
     command: /home/OXFORDNANOLABS/cwright/git/medaka/venv/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-fal177xf/biopython/setup.py'"'"'; __file__='"'"'/tmp/pip-install-fal177xf/biopython/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-c1oi4si4
         cwd: /tmp/pip-install-fal177xf/biopython/
    Complete output (1 lines):
    Biopython requires Python 3.6 or later. Python 3.5 detected.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

### PR DESCRIPTION
**Pull request by cjw85**
This pull request addresses issue #...

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [ ] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,7 +47,8 @@
 
 
 # Make sure we have the right Python version.
-if sys.version_info[:2] < (3, 6):
+PYVER = (3, 6)
+if sys.version_info[:2] < PYVER:
     sys.stderr.write(
         "Biopython requires Python 3.6 or later. "
```
Comment (peterjc) [inline setup.py:53]: The message needs to use ``PYVER`` too.

File: setup.py [hunk 2]
```diff
@@ -256,4 +257,5 @@ def can_import(module_name):
     ext_modules=EXTENSIONS,
     include_package_data=True,  # done via MANIFEST.in under setuptools
     install_requires=REQUIRES,
+    python_requires=">={}".format(".".join(str(x) for x in PYVER)),
```
Comment (peterjc) [inline setup.py:260]: From a style point of view, we're avoiding the format method and starting to use f-strings in the main code. However, setup.py is special as it is very helpful for it to work on older Python versions and run enough to give the version warning. So how about:

```python
python_requires=">=%i.%s" % PYVER,
```

File: setup.py (hunk 1 above)
Comment (peterjc) [reply setup.py:53]: Although see comment below, use and old fashioned percent formatting call so that ``setup.py`` works on old Python.

**Comment by cjw85** [comment]
I'll leave this with you.

**Comment by peterjc** [comment]
Would you like to be named in the NEWS or CONTRIB files?

**Comment by peterjc** [comment]
@MarkusPiotrowski could you review this please?

**Comment by peterjc** [comment]
@mdehoon could you review this instead?

**Comment by peterjc** [comment]
Thanks

**Pull request merged**